scenes/game/local_game.gd

```diff
@@ -1517,30 +1517,41 @@ class Player:
 	func discard(card_ids : Array):
 		var events = []
 		for discard_id in card_ids:
+			var found_card = false
+
 			# From hand
 			for i in range(len(hand)-1, -1, -1):
 				var card = hand[i]
 				if card.id == discard_id:
 					hand.remove_at(i)
 					events += add_to_discards(card)
 					on_hand_remove_public_card(discard_id)
+					found_card = true
 					break
+			if found_card: continue
 
 			# From gauge
 			for i in range(len(gauge)-1, -1, -1):
 				var card = gauge[i]
 				if card.id == discard_id:
 					gauge.remove_at(i)
 					events += add_to_discards(card)
+					found_card = true
 					break
+			if found_card: continue
 
 			# From overdrive
 			for i in range(len(overdrive)-1, -1, -1):
 				var card = overdrive[i]
 				if card.id == discard_id:
 					overdrive.remove_at(i)
 					events += add_to_discards(card)
+					found_card = true
 					break
+
+			if not found_card:
+				parent.printlog("ERROR: card to discard not found")
```

Comment:
maybe assert(false) too

Reply:
ooh i think this also secretly revealed a bug where multiple random discards could pick the same card twice